book/src/01-05-hello-triangle.md

```diff
@@ -0,0 +1,256 @@
+> Keep in mind that different GPU resources have specific alignment requirements that must be made explicit in your Rust types.
+
+Define an array of vertices to represent the triangle:
+
+```rust,no_run,no_playground
+const VERTICES: [Vertex; 3] = [
+    Vertex {
+        // bottom left
+        position: [-0.5, -0.5],
+        // red
+        color: [1.0, 0.0, 0.0],
+    },
+    Vertex {
+        // top
+        position: [0.0, 0.5],
+        // green
+        color: [0.0, 1.0, 0.0],
+    },
+    Vertex {
+        // bottom right
+        position: [0.5, -0.5],
+        // blue
+        color: [0.0, 0.0, 1.0],
+    },
+];
+```
+
+When uploading data to the GPU, `wgpu` expects a `&[u8]`.
+You can obtain this from a `&[T]` a couple ways:
+- Call `as_ptr()` on the slice, cast to `*const T` and then `*const u8`, and then reconstruct a `&[u8]` from the pointer with `slice::from_raw_parts`. This can be tricky if you're not confident about the particulars of `unsafe` code.
+- Use a library for performing this conversion, such as `bytemuck`. This is the approach taken by these examples, as it minimizes the amount of `unsafe` code being written.
+
+Add `bytemuck` to your `Cargo.toml`:
+
+```toml
+[dependencies]
+bytemuck = "1.3"
+```
+
+`bytemuck` provides a `Pod` ("plain old data") trait to mark types which can be safely cast to other `Pod` types, among other things.
+Since there are no special semantics to your `Vertex` type, you can safely `impl Pod` for it:
+
+```rust,no_run,no_playground
+use bytemuck::{Pod, Zeroable};
+
+// struct Vertex { ... }
+
+unsafe impl Zeroable for Vertex {}
+unsafe impl Pod for Vertex {}
+```
+
+`Pod` requires an `unsafe impl` because, when you `impl Pod` for a type, you agree to a contract stating that the type is safe to manipulate as raw bits and bytes without breaking something.
+The [`Pod` trait docs] have a more complete explanation of the contract.
+
+Now that `Vertex` can be easily cast to bytes, creating a vertex buffer is quite straightforward:
+
+```rust,no_run,no_playground
+use wgpu::util::DeviceExt;
+
+async fn run(event_loop: EventLoop<()>, window: Window) {
+    // let frag = ...
+    
+    // requires DeviceExt
+    let vertex_buffer = device.create_buffer_init(&wgpu::util::BufferInitDescriptor {
+        label: None,
+        contents: bytemuck::cast_slice(&VERTICES),
+        usage: wgpu::BufferUsage::COPY_DST | wgpu::BufferUsage::VERTEX,
+    });
+
+    // let size = ...
+}
+```
+
+This code uses `Device::create_buffer_init`, a convenience method provided by `wgpu` to initialize a buffer with data at creation.
+Since this method is not part of the WebGPU standard, it requires the `DeviceExt` trait to be imported.
+The `contents` of the buffer are provided with `bytemuck::cast_slice`, which cleanly converts a `&[Vertex]` to a `&[u8]`.
+The buffer requires the `COPY_DST` usage so that the provided data can be copied into it, and of course the `VERTEX` usage because it's a vertex buffer.
```

Comment:
COPY_DST isn't needed here. It's only needed when copying new data to the buffer after creating it. See https://docs.rs/wgpu/0.6.0/wgpu/struct.BufferUsage.html#associatedconstant.COPY_DST.